composition/content_fetcher.go

```diff
@@ -33,18 +27,25 @@ func (fr FetchResults) Less(i, j int) bool {
 	return fr[i].Def.Priority < fr[j].Def.Priority
 }
 
+// FetchDefinitionFactory should return a fetch definition for the given name and parameters.
+// This factory method can be used to supply lazy loaded fetch jobs.
+// The FetchDefinition returned has to have the same name as the supplied name parameter.
+type FetchDefinitionFactory func(name string, params Params) (fd *FetchDefinition, exist bool, err error)
```

Comment:
The meaning of the "exist" return value is unclear. Maybe add a comment describing its meaning. (Especially as there is another variable named "exist" in the AddFetchJob method)